### ISSUE TITLE
test code need!

### ISSUE THREAD

**Issue by BigCatHu** [fourson/Polarization-Aware-Low-Light-Image-Enhancement#2]

**Comment by BigCatHu** [comment]
We can't repeat the results in the paper, we didn't find the code for the test model in the repository, so we wrote a code for the test, the PSNR-p SSIM-p PSNR-θ SSIM-θ PSNR-S0 SSIM-S0 results are only 24.97, 0.67, 13.02, 0.26, 39.19, 0.97, and the results in the paper are 27.15 0.765 16.42 0.336 39.19 0.977, p and θ are quite different, the values of S0 match, can you provide us with a code for test so that we can reproduce the results in the paper

**Comment by fourson** [comment]
The dop and aop metrics are computed after average three channels. This is the similar way produced by IPLNet.

**Comment by BigCatHu** [comment]
> The dop and aop metrics are computed after average three channels. This is the similar way produced by IPLNet.

I used the calculation method you defined in the metric_utils, the average is used by default, you can provide the code for the test or I will send you the code for the test to correct the error.

```
def psnr(X, Y, data_range: float):
    """
    Peak Signal to Noise Ratio
    """

    mse = torch.mean((X - Y) ** 2)
    if mse == 0:
        return torch.tensor(50.)
    return 10 * torch.log10(data_range ** 2 / mse)


class PSNR(torch.jit.ScriptModule):
    __constants__ = ['data_range', 'avg']

    def __init__(self, data_range=1., avg=True):
        super().__init__()
        self.data_range = data_range
        self.avg = avg
        self.__name__ = 'PSNR'

    @torch.jit.script_method
    def forward(self, X, Y):
        r = psnr(X, Y, self.data_range)
        if self.avg:
            return r.mean()
        else:
            return r

```

**Comment by fourson** [comment]
Hello there,
Before using the PSNR defined in the metric_utils, I think you should do like this way:


        image_ref = np.load(ref_name)  # (H, W, 3) for RGB
        image_test = np.load(test_name)

        # transpose
        image_ref = np.expand_dims(np.transpose(image_ref, (2, 0, 1)), axis=0)  # (1, 3, H, W)
        image_test = np.expand_dims(np.transpose(image_test, (2, 0, 1)), axis=0)
        # to tensor
        image_ref_tensor = torch.from_numpy(image_ref).cuda().float()
        image_test_tensor = torch.from_numpy(image_test).cuda().float()

        # normalize
        if input_type == 'S0':
            image_ref_tensor = image_ref_tensor / 2
            image_test_tensor = image_test_tensor / 2
        elif input_type == 'AoP':
            image_ref_tensor = image_ref_tensor / np.pi
            image_test_tensor = image_test_tensor / np.pi

            # channle average
            image_ref_tensor = torch.mean(image_ref_tensor, dim=1, keepdim=True)
            image_ref_tensor = torch.cat([image_ref_tensor] * 3, dim=1)
            image_test_tensor = torch.mean(image_test_tensor, dim=1, keepdim=True)
            image_test_tensor = torch.cat([image_test_tensor] * 3, dim=1)
        else:
            # channle average
            image_ref_tensor = torch.mean(image_ref_tensor, dim=1, keepdim=True)
            image_ref_tensor = torch.cat([image_ref_tensor] * 3, dim=1)
            image_test_tensor = torch.mean(image_test_tensor, dim=1, keepdim=True)
            image_test_tensor = torch.cat([image_test_tensor] * 3, dim=1)

**Comment by fourson** [comment]
I think this issue could be closed. If you have further questions please create a new issue.
